lib/pbench/server/database/models/users.py

```diff
@@ -81,10 +81,9 @@ def roles(self):
     @roles.setter
     def roles(self, value):
         if isinstance(value, list):
-            for v in value:
-                self._roles += f";{v}"
+            self._roles = ";".join(value)
         elif isinstance(value, str):
-            self._roles = value if not self._roles else f"{self._roles};{value}"
+            self._roles = value
```

Comment:
FYI: one risk here (as before) is that this isn't sensitive to whether the incoming `str` is a single role or multiple roles and, if multiple, how they're separated. I'm not tracking down all the callers, but this isn't really ideal, and it's error prone.

Reply:
Oh the OIDC roles will never be `str` objects, they always come in the list. But now that you mentioned it, I realized that I have removed all the role update possibilities from the Server I should also remove this line, we can always expect the OIDC roles to come as the list object.

Reply:
What I meant is that the decoded payload would look something like this:
```
payload = {
        "iat": current_utc,
        "exp": exp,
        "sub": user.oidc_id,
        "aud": pbench-dashboard,
        "azp": pbench-dashboard,
        "realm_access": {
            "roles": [
                "default-roles-pbench-server",
                "offline_access",
                "uma_authorization",
            ]
        },
        "resource_access": {
            "broker": {"roles": ["read-token"]},
            "account": {
                "roles": ["manage-account", "manage-account-links", "view-profile"]
            },
            pbench-dashboard: {
                "roles": ["ROLE1", "ROLE2"]
            },
        },
        "scope": "openid profile email",
        "sid": "1988612e-774d-43b8-8d4a-bbc05ee55edb",
        "email_verified": True,
        "name": "first_name last_name",
        "preferred_username": username,
        "given_name": "first_name",
        "family_name": "last_name",
        "email": "dummy@example.com",
    }
```